### PR TITLE
Use new schema types for crypto-primitives and CIS-2 lib

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Update types to use the new LEB128 and ByteArray when it makes sense.

## Changes

- CIS2 Token IDs are now ByteLists.
- CIS2 Token amounts are now ULEB128 constraint to 37 bytes.
- Crypto primitives are now ByteArrays.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-cis2/src/lib.rs
Comment (abizjak) [inline concordium-cis2/src/lib.rs:502]: Please explain the 37 constant in the comments.

**Review by abizjak — COMMENTED** [review]
This looks good, but needs to wait for submodule update for approval.

File: concordium-cis2/src/lib.rs
Comment (limemloh) [reply concordium-cis2/src/lib.rs:504]: I have added a comment

File: concordium-std/CHANGELOG.md
Comment (limemloh) [reply concordium-std/CHANGELOG.md]: I have extended this

File: concordium-std/CHANGELOG.md
Comment (Bargsteen) [inline concordium-std/CHANGELOG.md:3]:
```suggestion
- Change SchemaType implementation for cryptographic primitives to ByteArray, meaning that the primitives(e.g., hashes and signatures) are now supplied as hex strings in JSON.
```

**Review by Bargsteen — APPROVED** [review]

**Review by abizjak — APPROVED** [review]

**Pull request merged**
